sklearn/compose/_target.py

```diff
@@ -162,7 +162,7 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
-        y = check_array(y, accept_sparse=False, force_all_finite=True,
+        y = check_array(y, accept_sparse=False, force_all_finite='allow-nan',
```

Comment:
I'd consider going further to switch off all finiteness validation. Are there cases where it would be risky to pass an inf through unchecked?

Reply:
I guess it would be safe to leave both INF and NAN checks to the actual transformer. I'll give it a go, let's check if it passes the tests.